src/Http/Routing/src/Builder/DelegateEndpointRouteBuilderExtensions.cs

```diff
@@ -231,6 +182,75 @@ public static class DelegateEndpointRouteBuilderExtensions
             return new DelegateEndpointConventionBuilder(dataSource.AddEndpointBuilder(builder));
         }
 
+        /// <summary>
+        /// Adds a <see cref="RouteEndpoint"/> to the <see cref="IEndpointRouteBuilder"/> that matches HTTP requests
+        /// for the specified HTTP methods and pattern.
+        /// </summary>
+        /// <param name="endpoints">The <see cref="IEndpointRouteBuilder"/> to add the route to.</param>
+        /// <param name="pattern">The route pattern.</param>
+        /// <param name="handler">The delegate executed when the endpoint is matched.</param>
+        /// <param name="httpMethods">HTTP methods that the endpoint will match.</param>
+        /// <returns>A <see cref="IEndpointConventionBuilder"/> that can be used to further customize the endpoint.</returns>
+        public static DelegateEndpointConventionBuilder MapMethods(
+           this IEndpointRouteBuilder endpoints,
+           string pattern,
+           IEnumerable<string> httpMethods,
+           Delegate handler)
+        {
+            return MapMethods(endpoints, pattern, httpMethods, handler, allowImplicitFromBody: true);
```

Comment:
We were discussing preventing implicit from-body inference even for MapMethods() calls if the `httpMethods` only contains a non-empty subset of GET/DELETE/OPTIONS/HEAD. Any reason not to?

Reply:
> I didn't do any http method matching to set the AllowImplicitFromBody property to simplify things

I'm not against doing it, it's just ugly code I didn't want to write while working on the main part of the change 😆

Reply:
I think that's problematic and we shouldn't do it. It's Similar to how we can't protect this in app.Map (any method).